openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -0,0 +1,114 @@
+                        code developed for ETL. The compromise here is a lot of SQL wrapped in Python. 
+                        
+                        Another approach to consider would be an open source tool developed specifically for 
+                        SQL transformations -- dbt. Reading this article -- https://docs.getdbt.com/blog/dbt-airflow-spiritual-alignment
+                        -- made me wonder if a proof of concept with Inaturalist data might be worthwhile.
+                        dbt core is fully open source, and assuming that there will be more ELT providers, 
+                        it could be worthwhile.
+
+Output:                 This had been "TSV file containing the media metadata." but I'm not 100% sure that 
+                        that makes sense given the whole ELT vs ETL thing.
+
+Notes:                  The inaturalist API is not intended for data scraping.
+                        https://api.inaturalist.org/v1/docs/
+                        But there is a full dump intended for sharing on S3.
+                        https://github.com/inaturalist/inaturalist-open-data/tree/documentation/Metadata
+                        Because these are very large normalized tables, as opposed to more document oriented API 
+                        responses, we found that bringing the data into postgres first was the most effective approach.
+                        More detail in slack here:
+                        https://wordpress.slack.com/archives/C02012JB00N/p1653145643080479?thread_ts=1653082292.714469&cid=C02012JB00N
+                        This uses the structure defined here, except for adding ancestry tags to the taxa table:
+                        https://github.com/inaturalist/inaturalist-open-data/blob/main/Metadata/structure.sql
+
+TO DO:                  There is nothing here that actually updates the images table.
+                        Need to figure out a stand-in for metrics from the saved json counter
+
+"""
+import json
+import os
+import logging
+from pathlib import Path
+from urllib.parse import urlparse
+
+from textwrap import dedent
+import psycopg2
+
+from common.loader import provider_details as prov
+from common.storage.image import ImageStore
+
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+PROVIDER = prov.INATURALIST_DEFAULT_PROVIDER
+SCRIPT_DIR = '/usr/local/airflow/openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/'
+
+# This is set up here as a single thread linear process, to be more consisten with the 
+# structure of the provider dag factory, but if I were to start from scratch with airflow, 
+# each of these would be a task, and the dag would be 00 >> [01, 02, 03, 04] >> 05.
+# In dbt, the schema would be created separately, and steps 1-5 would each be a "model"
```

Comment:
Good call! You could definitely manage this using @AetherUnbound's [suggestion](https://github.com/WordPress/openverse-catalog/pull/549#pullrequestreview-1015851095) to take the DAG and build a TaskGroup that can be set as an upstream task dependency.

Reply:
Is it possible that if I did everything with Airflow hooks the boto issue I'm seeing would go away? I vaguely remember that airflow also has an s3 --> postgres method somewhere.